### PR TITLE
CI -- only run fuzz tests on ubuntu

### PR DESCRIPTION
**Pull request by danlaine**
## Why this should be merged

There's no reason to run the fuzz test on every build type. We can just do it on the fastest one.

## How this works

* Don't run fuzz tests in `build_and_test.sh`
* Only run them on linux arm64 runner

## How this was tested

Look at CI output

### CODE REVIEW

File: .github/workflows/build_and_test.sh [hunk 1]
```diff
@@ -15,5 +15,6 @@ if [[ -z $(git status -s) ]]; then
     # TODO: Revise this check once we can reliably build without changes
     # exit 1
 fi
+
+# Note this doesn't run the fuzz tests in scripts/build_fuzz.sh
```
Comment (StephenButtolph) [inline .github/workflows/build_and_test.sh:19]: This comment seems unnecessary... It makes sense as part of the PR but doesn't make much sense in the code imo.

File: .github/workflows/build-and-test.yml [hunk 2]
```diff
@@ -25,3 +25,9 @@ jobs:
       - name: build_test
         shell: bash
         run: .github/workflows/build_and_test.sh
+      - name: fuzz_test
+        shell: bash
+        if: matrix.os[0] == 'self-hosted' && matrix.os[3] == 'jammy' # Only run on self-hosted jammy
```
Comment (StephenButtolph) [inline .github/workflows/build-and-test.yml:30]: Is there a reason why we run the fuzz testing on `arm` instances?
```suggestion
        if: matrix.os == 'ubuntu-22.04'
```
Guess they might be faster

Comment (danlaine) [reply .github/workflows/build-and-test.yml:30]: yeah I just picked the fastest one

**Review by StephenButtolph — APPROVED** [review]

**Comment by patrick-ogrady** [comment]
I actually recommend running this on `ubuntu-latest` (which is free for OSS repos): https://docs.github.com/en/billing/managing-billing-for-github-actions/about-billing-for-github-actions#about-billing-for-github-actions

**Comment by danlaine** [comment]
> I actually recommend running this on `ubuntu-latest` (which is free for OSS repos): https://docs.github.com/en/billing/managing-billing-for-github-actions/about-billing-for-github-actions#about-billing-for-github-actions
> 
> You can also run it for longer (again because it is free).

done

**Review by abi87 — APPROVED** [review]
lgtm

**Pull request merged**